### PR TITLE
Integrate llvm-project and bump dependencies (2022-10-13)

### PR DESCRIPTION
**Pull request by kuhar**
* llvm-project: 7ac10e5b347f3e22ae1dad5937de667e57c472be
* mlir-hlo: e8b5c0d438d4de258b6239b5cbd9798dbe652305
* tensorflow: 2c18ea76a335f3120076568df1a3eaa129678b4c

### CODE REVIEW

File: compiler/src/iree/compiler/InputConversion/MHLO/test/mhlo_to_mhlo_preprocessing.mlir [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // RUN: iree-opt --split-input-file --verify-diagnostics --iree-mhlo-to-mhlo-preprocessing %s | FileCheck %s
+// XFAIL: *
```
Comment (hanhanW) [inline compiler/src/iree/compiler/InputConversion/MHLO/test/mhlo_to_mhlo_preprocessing.mlir:2]: do we have to disable all the tests? I thought it's a failure about scatter op?

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/InputConversion/MHLO/test/mhlo_to_mhlo_preprocessing.mlir:2]: See Discord. I think this is just for testing... Please make sure this is removed before the integrate lands, and only the failing test is disabled (if it isnt root caused that is).

Comment (hanhanW) [reply compiler/src/iree/compiler/InputConversion/MHLO/test/mhlo_to_mhlo_preprocessing.mlir:2]: I see, sorry for the spamming

Comment (kuhar) [reply compiler/src/iree/compiler/InputConversion/MHLO/test/mhlo_to_mhlo_preprocessing.mlir]: I updated the PR to only disable the failing tests. I also added a FIXME comment with the link to the issue: https://github.com/iree-org/iree/issues/10774

**Review by hanhanW — APPROVED** [review]

**Review by dcaballe — APPROVED** [review]

**Pull request merged**